dockerfiles/alpine/apim-universal-gw/docker-entrypoint.sh

```
@@ -54,6 +54,18 @@ test -d ${config_volume} && [[ "$(ls -A ${config_volume})" ]] && cp -RL ${config
+stop_handler() {
+  echo "Stopping WSO2 gracefully..." >&2
+  sh "${WSO2_SERVER_HOME}/bin/gateway.sh" stop
+  wait "${server_pid}"
+}
+
+trap 'stop_handler' SIGTERM SIGINT
+
 # start WSO2 Carbon server
 echo "Start WSO2 Carbon server" >&2
-sh ${WSO2_SERVER_HOME}/bin/gateway.sh "$@"
+sh ${WSO2_SERVER_HOME}/bin/gateway.sh "$@" &
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)